### ISSUE TITLE
Clicking on collapsed debug session stack expands then collapses it immediately

### ISSUE THREAD

**Issue by roblourens** [Microsoft/vscode#64463]
- Just debugging two normal sessions
- A is active, B is collapsed
- Click B, it blinks open then collapses again

I'm guessing it's toggled by the click and by becoming active and they interfere?

![dec-05-2018 15-09-01](https://user-images.githubusercontent.com/323878/49550169-b978f280-f89f-11e8-9c41-6d29dead0290.gif)

**Comment by isidorn** [comment]
The issue here is the following:
There is a tree navigator, which will react on click -> based on that reaction I will call tree.expand
However while I am doing this the tree plans to toggle expansion because of the click. My expand gets resovled first, which expands the element and than the tree collapses it.

@joaomoreno on advice what best to do here

**Comment by joaomoreno** [comment]
> There is a tree navigator, which will react on click -> based on that reaction I will call tree.expand

Why are we calling expand there?

**Comment by isidorn** [comment]
Because of the way my code is strucutred, whenever a stack frame gets focused I want to expand the session and the thread to show the stack frame.

So what happens here is a weird cycle. Click on sessions, session due to navigator gets focused, because session gets focused we find a stack frame belonging to it and it gets focus and the expand happens

**Comment by roblourens** [comment]
The workaround of keeping a flag on the callstack's side is awkward and would be nicer if we could pass some context through the tree like the old tree allows.

**Comment by joaomoreno** [comment]
New tree allows that too.

**Comment by isidorn** [comment]
Does not apply in this use case.
But thanks for the advice, however for now I will go with the workaround.

**Comment by roblourens** [comment]
The browserEvent? https://github.com/Microsoft/vscode/blob/master/src/vs/base/browser/ui/tree/abstractTree.ts#L361 For search, I'm looking for a way to pass some arbitrary flag in, although I could pack it into a fake browser event.

**Comment by joaomoreno** [comment]
Yeah the browser event... I'm not too keen on allowing `any` there. What is the use case exactly for search?

**Comment by roblourens** [comment]
I opened https://github.com/Microsoft/vscode/issues/64743
